packages/core/src/component.ts

```diff
@@ -23,17 +30,46 @@ export interface Component {
    * An array of controller classes
    */
   controllers?: ControllerClass[];
+
   /**
-   * A map of name/class pairs for binding providers
+   * A map of providers to be bound to the application context
+   * * For example:
+   * ```ts
+   * {
+   *   'authentication.strategies.ldap': LdapStrategyProvider
+   * }
+   * ```
    */
   providers?: ProviderMap;
+
+  /**
+   * A map of classes to be bound to the application context.
+   *
+   * For example:
+   * ```ts
+   * {
+   *   'rest.body-parsers.xml': XmlBodyParser
+   * }
+   * ```
+   */
+  classes?: ClassMap;
+
   /**
    * A map of name/class pairs for servers
    */
   servers?: {
     [name: string]: Constructor<Server>;
   };
 
+  /**
+   * An array of bindings to be aded to the application context. For example,
+   * ```ts
+   * const bindingX = new Binding('x').to('Value X');
```

Comment:
Let's use the newly added sugar API please.
```suggestion
   * const bindingX = Binding.bind('x').to('Value X');
```